mibitracker/request_helpers.py

```diff
@@ -61,13 +62,30 @@ class MibiRequests():
             response's status code is >= 400.
     """
 
-    def __init__(self, url, email, password, retries=MAX_RETRIES,
+    def __init__(self,
+                 url,
+                 email=None,
+                 password=None,
+                 token=None,
+                 retries=MAX_RETRIES,
                  retry_methods=RETRY_METHOD_WHITELIST,
                  retry_codes=RETRY_STATUS_CODES):
 
         self.url = url.rstrip('/')  # We add this as part of request params
         self.session = StatusCheckedSession()
-        self._auth(url, email, password)
+
+        # Provide either an email and password, or a token
+        # The token will be used in lieu of email and password if provided
+        if token is not None:
+            self.session.headers.update({
+                'Authorization': 'JWT {}'.format(token)
+            })
```

Comment:
This looks great but as you just mentioned it'd be even better if you validated the token here too.